### ISSUE TITLE
JAWS does not announce aria-expanded on a native button element if aria-haspopup is present (regardless of its boolean/token value).

### ISSUE THREAD

**Issue by electronicwoft** [FreedomScientific/VFO-standards-support#211]
JAWS does not announce aria-expanded on a native button element if aria-haspopup is present (regardless of its boolean/token value).

1. create simple native buttone element with aria-expanded 
2. create simple native buttone element with aria-expanded and aria-haspopup
3. launch JAWS with default switch. Launch example in a windows browser
4. navigate to buttons using tab or quickkeys
5. note that buttons with aria-haspopup do not announce expand state of button

Expected result is that name, role, state, and property are announced when button is navigated to using tab or quickkey

The actual result is that JAWS only announces name and 'menubutton' role

**Comment by stevefaulkner** [comment]
@electronicwoft Can you please provide a working example (codepen or other) that exhibits the issue. Thanks!

**Comment by electronicwoft** [comment]
[https://electronicwoft.github.io/](url) .... might be a Chrome issue? in any case, shouldn't this state be announced when a control is focused using the tab key?

**Comment by JAWS-test** [comment]
This is a long-standing bug in JAWS: buttons with aria-haspopup are output as menus and not as buttons. The missing output of aria-expanded is just a consequence. However, the error is not due to JAWS, but is an error of the specification that specifies for aria-haspopup: "If on a push button, change the role to ROLE_SYSTEM_BUTTONMENU" (https://www.w3.org/TR/wai-aria-implementation/)

**Comment by electronicwoft** [comment]
Hi @JAWS-test. Thanks for your response. So is it JAWS or the specification or the browser(s) that is/are in error when aria-expanded is announced correctly when aria-haspopup is present on a native button (eg: Firefox)?? Why is aria-expanded not being announced by JAWS in Chrome a consequence of changing from pushbutton to menubutton?

**Comment by electronicwoft** [comment]
oh, and why is aria-expanded not announced by JAWS in ANY browser when aria-haspopup is present and when a button is tabbed to?

**Comment by JAWS-test** [comment]
Hi @electronicwoft: In my opinion the initial bug is the wrong W3C specification regarding buttons with aria-haspopup. Unfortunately Freedom Scientific took over the bug in JAWS. Correct would be, if aria-haspopup is output the same at all elements: "Has Pop-up". Unfortunately, buttons with aria-haspopup will output  as a menu. My recommendation would be to bypass the bug by using links instead of buttons. An alternative possibility would be to do without aria-haspopup and mark the buttons only with aria-expanded. This is a sufficient indication that something can be opened via the button. Actually the use of aria-haspopup and aria-expanded is redundant information.

**Comment by electronicwoft** [comment]
Hi @JAWS-test. what is the difference between 'button collapsed' or 'link collapsed' and 'menubutton collapsed'? IMHO, the expanded state of a control is not redundant information. If it is, this would be completely inconsistent. The role exposed by an AAPI is not what is announced by a screen reader, anyway. Unfortunately, I can't 'just use links' to workaround what is a defect in either JAWS and/or Chrome. If you believe that the role mapping is incorrect for this attribute, then bug the relevant specification. Otherwise, the issue described above should be addressed by either JAWS or Chrome.

**Comment by JAWS-test** [comment]
Hi @electronicwoft: aria-haspopup is redundant if there is aria-expanded. aria-haspopup informs that there is a page area that can be faded in and out. The same is done by aria-expanded, but with the advantage that aria-expanded also tells whether the area is currently faded in or out. I.e. if aria-haspopup is omitted, no information will be lost.
The bug cannot be fixed by Chrome, since it occurs independently of the browser. It's a JAWS problem, but it's due to the specification. So it should be fixed by JAWS and the W3C.

**Comment by scottaohara** [comment]
test case: https://codepen.io/anon/pen/pXYPYa

**Comment by electronicwoft** [comment]
Hi @JAWS-test ... um, yeah, nah. with respect, can I recommend you return to the ARIA specification and revisit the roles model and supported states and properties? and, by the way, do you represent Freedom Scientific or FVO in any way?

**Comment by electronicwoft** [comment]
Hi @scottaohara. Thanks for the codepen - if you copy and paste the hyperlink I provided further up, there is an example with test results.

**Comment by scottaohara** [comment]
A specific test case means people don’t need to hunt for it though.

**Comment by electronicwoft** [comment]
... just that codepen is a pain with a screen reader, but, yes, appreciate its utility.

**Comment by JAWS-test** [comment]
Hi @electronicwoft, I don't represent JAWS or anyone else. Sorry for any misunderstanding. I just have to work with JAWS every day.

**Comment by stevefaulkner** [comment]
@JAWS-test I created a more exhaustive test case https://s.codepen.io/stevef/debug/QXoEpx this is a debug view codepen and should not cause you any issue as a screenreader user.

**Comment by JAWS-test** [comment]
using https://s.codepen.io/stevef/debug/QXoEpx  and JAWS 2019 / Windows 10:

IE 11, Firefox: Tab navigation: status aria-expanded is not output if aria-haspopup, linear reading: Status aria-expanded is always output

Chrome: Status aria-expanded is not output on tab navigation and linear reading if aria-haspopup

Edge: Status aria-expanded is always output on tab navigation and linear reading if aria-haspopup (because this does not lead to wrong output as menu)

**Comment by electronicwoft** [comment]
Hi @stevefaulkner - thanks for the updated pen.
With Chrom 75, Windows 7, & JAWS2019:
using quickkey navigation:
1. expanded false Button collapsed 
2. expanded true Button expanded 
3.  has popup button menu 
4. expanded false has popup button menu 
5. expanded true has popup button menu
and using tab navigation:
1. expanded false Button collapsed 
2. expanded true Button expanded 
3.  has popup button menu 
4. expanded false has popup button menu 
5. expanded true has popup button menu

**Comment by JAWS-test** [comment]
@electronicwoft https://github.com/w3c/core-aam/issues/51
